### PR TITLE
Test against EO compiler

### LINKED ISSUE #100: Test against EO compiler
Supersedes #3, #93, #46

- reuse tests from the eo repository using submodules
- add a eo subtree to normalizer
- write scripts that prepare a subset of tests
- adapt existing scripts that run eo tests
- set up a pipeline job

### PR DESCRIPTION
**Pull request by deemp**
Closes #100, #3, #46

Supersedes #95

Blocked on
- #106
- https://github.com/objectionary/eo/issues/2868
- #101
- https://github.com/objectionary/eo/issues/2873
- #153

Non-critical issues
- #180
- https://github.com/freckle/stack-action/issues/56
- https://github.com/objectionary/eo/issues/2930

- [x] Reuse tests from the eo repository using submodules

---

## PR-Codex overview
This PR updates dependencies, adds a new submodule, modifies scripts for pipeline testing, and refactors the project structure.

### Detailed summary
- Updated `eolang` dependency to `^0.18.0`
- Added `transform-eo-tests` submodule
- Modified scripts for `pipeline` testing
- Refactored project structure
- Updated `stack.yaml` and `.gitignore` files
- Updated `hie.yaml` and `scripts/transform-eo-tests/package.yaml`
- Added new scripts `try-unphi.sh` and `pipeline.sh`
- Updated `transform-eo-tests.cabal`
- Updated workflows in `.github` folder
- Updated `flake.nix` and `pipeline/config.yaml` files

> The following files were skipped due to too many changes: `pipeline/config.yaml`, `scripts/transform-eo-tests/app/Main.hs`, `package-lock.json`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: scripts/transform-eo-tests/package.yaml
Comment (aabounegm) [inline scripts/transform-eo-tests/package.yaml:1]: Why is it a separate package rather than just another executable in the same package?

Comment (deemp) [reply scripts/transform-eo-tests/package.yaml:1]: For now, it has nothing to do with the eo-phi-normalizer. It's just a convenience script.

**Review by fizruk — APPROVED** [review]

**Comment by 0pdd** [comment]
@deemp the puzzle [#250](https://github.com/objectionary/normalizer/issues/250) is still not solved.

**Comment by 0pdd** [comment]
@deemp the only puzzle [#250](https://github.com/objectionary/normalizer/issues/250) is solved here.

**Pull request merged**